Zadanie_1_AES/main.cpp

```diff
@@ -1,28 +1,132 @@
 #include <iostream>
+#include <cstdio>
+#include <vector>
+#include <cstring>
 #include "AES.h"
 
-int main() {
+#define BUFOR_SIZE  (16)
 
-    unsigned char plainText[] = { 0x32, 0x43, 0xf6, 0xa8,
-                                  0x88, 0x5a, 0x30, 0x8d,
-                                  0x31, 0x31, 0x98, 0xa2,
-                                  0xe0, 0x37, 0x07, 0x34 };
+int fromhex(char val) {
+    if (val >= '0'  && val <= '9') {
+        return val - '0';
+    }
+    if (val >= 'a' && val <= 'f') {
+        return val - 'a' + 10;
+    }
+    printf("Zmyslny komunikat");
+    exit(8);
+}
 
-    unsigned char key[] =    { 0x2b, 0x7e, 0x15, 0x16,
-                               0x28, 0xae, 0xd2, 0xa6,
-                               0xab, 0xf7, 0x15, 0x88,
-                               0x09, 0xcf, 0x4f, 0x3c};
 
-    unsigned char result[] = {0x39, 0x02, 0xdc, 0x19,
-                              0x25, 0xdc, 0x11, 0x6a,
-                              0x84, 0x09, 0x85, 0x0b,
-                              0x1d, 0xfb, 0x97, 0x32 };
 
-    AES aes(4, 4, 10);
-    aes.Encrypt(plainText,key);
-    printf("\nZakonczono szyfrownie.\n");
-    aes.Decrypt(result, key);
+int main(int argc, char* argv[]) {
 
+    std::string inputFileNameString;
+    std::string outputFileNameString;
+    std::string keyString;
+    char* inputFileName;
+    char* outputFileName;
+    FILE* inputFile;
+    FILE* ouputFile;
+    std::vector<unsigned char> bufor(BUFOR_SIZE);
+    char choice;
+    unsigned char key[16];
+
+    if ((argc != 4) && (argc != 5)) {
+        printf ("Niepoprawne argumenty programu.\n");
```

Comment:
`printf `and other _variadic functions_ can be reason of **type-safety** troubles, thus should not be used in new code. Use `std::cout` or `puts`(cleaner assembly code, less readability)